libraries/AP_IBus_Telem/AP_IBus_Telem.h

```diff
@@ -0,0 +1,355 @@
+    } _sensor_definition;
+
+    typedef struct {
+        uint8_t ap_mode;
+        uint8_t ibus_mode;
+    } _mode_map;
+
+    void ensure_listening();
+    void reset_read_state_after_timegap();
+    void handle_incoming_message();
+    void handle_discover_command();
+    void handle_type_command();
+    void handle_value_command();
+    void handle_2_byte_value_command(uint8_t sensor_type, _sensor_value value);
+    void handle_4_byte_value_command(uint8_t sensor_type, _sensor_value value);
+    _sensor_definition* get_sensor_definition(uint8_t sensor_id);
+    _sensor_value get_sensor_value(uint8_t sensor_id);
+    uint16_t get_average_cell_voltage_cV();
+    uint16_t get_current_cAh();
+    uint8_t get_battery_or_fuel_level_pct();
+    uint16_t get_rpm();
+    uint8_t get_gps_status();
+    Vector3f get_unbiased_acceleration();
+    uint16_t get_distance_from_home_m();
+    uint16_t get_vehicle_mode();
+    uint16_t get_mapped_vehicle_mode(_mode_map *vehicle_mode_map, uint8_t map_size);
+    void populate_checksum(uint8_t *packet, uint16_t size);
+
+    enum class ReadState {
+        READ_LENGTH,
+        READ_COMMAND,
+        READ_CHECKSUM_LOW_BYTE,
+        READ_CHECKSUM_HIGH_BYTE,
+        DISCARD,
+    };
+
+    static const uint8_t PROTOCOL_COMMAND_DISCOVER = 0x80; // Command to discover a sensor (lowest 4 bits are sensor)
+    static const uint8_t PROTOCOL_COMMAND_TYPE = 0x90;     // Command to request a sensor type (lowest 4 bits are sensor)
+    static const uint8_t PROTOCOL_COMMAND_VALUE = 0xA0;    // Command to request a sensor's value (lowest 4 bits are sensor)
+    static const uint8_t PROTOCOL_TIMEGAP = 3;             // Packets are received very ~7ms so use ~half that for the gap
+    static const uint8_t PROTOCOL_FOUR_LENGTH = 0x04;      // indicate that the message has 4 bytes
+    static const uint8_t PROTOCOL_SIX_LENGTH = 0x06;       // indicate that the message has 6 bytes
+    static const uint8_t PROTOCOL_EIGHT_LENGTH = 0x08;     // indicate that the message has 8 bytes
+    static const uint8_t PROTOCOL_INCOMING_MESSAGE_LENGTH = PROTOCOL_FOUR_LENGTH; // All incoming messages are the same length
+
+    ReadState _read_state = ReadState::READ_LENGTH;
+    uint32_t _last_received_message_time_ms; // When the last message was received
+    uint8_t _incoming_command;               // The command coming from the receiver
+    uint8_t _incoming_sensor_id;             // The sensor the receiver is talking to us about
+    uint16_t _expected_checksum;             // Checksum calculation
+    uint8_t _incoming_checksum_high_byte;    // Checksum high byte storage
+
+    // All the sensors we can accurately provide are listed here.
+    // i-BUS will generally only query up to 15 sensors, so subjectively
+    // higher-value sensors are sorted to the top to make the most of a
+    // small telemetry window. In the future these could be configurable.
+    _sensor_definition _sensors[25] = {
```

Comment:
```suggestion
    static const SensorDefinition _sensors[25] = {
```

similarly for the other definitions.

These currently consume RAM, which is bad.  `static const` moves it to just flash.

Reply:
I'm being told I can't initialise these here without also adding `constexpr`; that seems to be working but let me know if there's a preferred way.

Reply:
I notice you changed the case for this struct type, so I've applied that to the other struct types as well. Easy to change it back to snake case if that was accidental.